app/[locale]/[...uriSegments]/page.tsx

```diff
@@ -0,0 +1,176 @@
+import { OpenGraph } from "next/dist/lib/metadata/types/opengraph-types";
+import striptags from "striptags";
+import { getBreadcrumbsById, getEntryMetadataByUri } from "@/lib/api/metadata";
+import {
+  getEntriesByLocale,
+  getEntrySectionByUri,
+} from "@/lib/api/entries/index";
+import { getEntryDataByUri } from "@/lib/api/entry";
+import { getReleaseOpenGraph } from "@/lib/api/noirlab";
+import { resizeCantoImage } from "@/lib/api/canto/resize";
+import PageTemplate from "@/components/templates/Page";
+import NewsPageTemplate from "@/components/templates/NewsPage";
+import GlossaryPageTemplate from "@/components/templates/GlossaryPage";
+import SearchPageTemplate from "@/components/templates/SearchPage";
+import SlideshowPageTemplate from "@/components/templates/SlideshowPage";
+import StaffPageTemplate from "@/components/templates/StaffPage";
+import EventPageTemplate from "@/components/templates/EventPage";
+
+const pickFeaturedImage = async (
+  locale: string,
+  image?: any,
+  cantoAsset?: any,
+  pressReleaseId?: string
+): Promise<OpenGraph["images"] | undefined> => {
+  if (pressReleaseId) {
+    const releaseOpenGraphImage = await getReleaseOpenGraph(
+      locale,
+      pressReleaseId
+    );
+
+    return releaseOpenGraphImage;
+  }
+
+  if (cantoAsset) {
+    const {
+      width,
+      height,
+      url: { directUrlPreview },
+    } = cantoAsset;
+    const url = resizeCantoImage(directUrlPreview, 800);
+
+    return { url, width, height };
+  }
+
+  if (image) {
+    const { url, width, height, altText: alt } = image;
+
+    return { url, width, height, alt };
+  }
+};
+
+export async function generateMetadata(
+  { params: { locale, uriSegments }, searchParams = {} }: UriSegmentProps,
+  parent: ResolvingMetadata
+): Promise<Metadata> {
+  const uri = uriSegments.join("/");
+  let previewToken: string | undefined;
+
+  if (draftMode().isEnabled) {
+    previewToken = Array.isArray(searchParams.preview)
+      ? searchParams.preview[0]
+      : searchParams?.preview;
+  }
+
+  const { entry } = await getEntryMetadataByUri(uri, locale, previewToken);
+
+  if (!entry) {
+    notFound();
+  }
+
+  const {
+    title,
+    description,
+    image = [],
+    cantoAssetSingle = [],
+    pressReleaseId,
+  } = entry;
+
+  const featuredImage = await pickFeaturedImage(
+    locale,
+    image[0],
+    cantoAssetSingle[0],
+    pressReleaseId
+  );
+  const previousImages = (await parent).openGraph?.images || [];
+
+  return {
+    title,
+    description: striptags(description),
+    openGraph: {
+      images: featuredImage || previousImages,
+    },
+  };
+}
+
+export async function generateStaticParams({
+  params: { locale },
+}: LocaleProps) {
+  const data = await getEntriesByLocale(locale);
+
+  return data.entries.map(({ uri }) => {
+    return {
+      locale,
+      uriSegments: uri.split("/"),
+    };
+  });
+}
+
+const sectionMap = {
+  events: EventPageTemplate,
+  // galleryItems: GalleryPageTemplate,
```

Comment:
for now I am excluding the gallery and user profile pages since both are unused.